lib/SIL/Utils/BasicBlockUtils.cpp

```diff
@@ -416,6 +416,17 @@ void JointPostDominanceSetComputer::findJointPostDominatingSet(
   // always clean up any resources that we use!
   SWIFT_DEFER { clear(); };
 
+  /// A set that guards our worklist. Any block before it is added to worklist
+  /// should be checked against visitedBlocks.
+  SILFunction *function = dominatingBlock->getParent();
+  BasicBlockSet visitedBlocks(function);
+
+  /// The set of blocks where we begin our walk.
+  BasicBlockSet initialBlocks(function);
+
+  /// True for blocks which are in blocksThatLeakIfNeverVisited.
+  BasicBlockFlag isLeakingBlock(function);
```

Comment:
Now that we are no longer removing elements from blocksThatLeakIfNeverVisited, why not change that to be a BasicBlockSetVector? Then we can eliminate this set and can just use blocksThatLeakIfNeverVisited, right?

Reply:
Erik and I spoke about this and he mentioned that the reason why he didn't do that is he was worried about algorithmic changes in how the algorithm worked in this case. I am going to think about that and if we change this it will be in a follow on commit.